packages/packages/store/src/__tests__/store.test.tsx

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import { renderHook } from '@testing-library/react-hooks';
+import {
+	createStoreService,
+	StoreProvider,
+	useSelector,
+	useDispatch,
+	Dispatch,
+	AnyAction,
+} from '../index';
+
+interface SliceStateRoot {
+	slice: {
+		value?: number;
+	}
+}
+
+interface Config {
+	initialValue?: SliceStateRoot['slice']['value'];
+}
+
+const createStoreEntities = ( storeService: ReturnType<typeof createStoreService>, { initialValue = 1 }: Config = {} ) => {
+	const slice = storeService.registerSlice( {
+		name: 'slice',
+		initialState: {
+			value: initialValue,
+		},
+		reducers: {
+			setValue: ( state, action ) => {
+				state.value = action.payload;
+			},
+		},
+	} );
+
+	const store = storeService.createStore();
+
+	const wrapper: React.FC = ( { children } ) => (
+		<StoreProvider store={ store }>
+			{ children }
+		</StoreProvider>
+	);
+
+	return {
+		slice,
+		store,
+		wrapper,
+	};
+};
+
+describe( '@elementor/store', () => {
+	it( 'should verify the initial state of the slice', () => {
+		const storeService = createStoreService();
+
+		const { wrapper } = createStoreEntities( storeService );
+
+		const { result } = renderHook( () => useSelector( ( state: SliceStateRoot ) => state.slice.value ), { wrapper } );
+
+		expect( result.current ).toBe( 1 );
+	} );
+
+	it( 'should update the state value of the slice', () => {
+		const storeService = createStoreService();
+
+		const { slice, wrapper } = createStoreEntities( storeService );
+
+		const { result } = renderHook( () => {
+			const dispatchAction = useDispatch();
+
+			dispatchAction( slice.actions.setValue( 3 ) );
+
+			return useSelector( ( state: SliceStateRoot ) => state.slice.value );
+		}, { wrapper } );
+
+		expect( result.current ).toBe( 3 );
+	} );
+
+	it( 'should throw an error when trying to register a slice with the same name more than once', () => {
+		const storeService = createStoreService();
+
+		storeService.registerSlice( {
+			name: 'slice',
+			initialState: {
+				value: 1,
+			},
+			reducers: {},
+		} );
+
+		expect( () => {
+			storeService.registerSlice( {
+				name: 'slice',
+				initialState: {
+					value: 1,
+				},
+				reducers: {},
+			} );
+		} ).toThrowError();
```

Comment:
maybe assert the thrown string?
`.toThrow( 'slice already exists' )`

Reply:
Done.